render/src/lib/views/sidebar.vertical/port.available/components.ts

```diff
@@ -0,0 +1,304 @@
+// tslint:disable:no-inferrable-types
+
+import { Component, OnDestroy, Input, AfterViewInit, ViewChildren, QueryList, ElementRef, ChangeDetectorRef, OnInit } from '@angular/core';
+import { IPortInfo } from '../../../common/interface.portinfo';
+import Chart from 'chart.js';
+import { Subscription, Subject } from 'rxjs';
+import Service, { IPort } from '../../../services/service';
+import { EHostEvents } from '../../../common/host.events';
+import { SidebarVerticalPortDialogComponent } from '../port.options/component';
+import { IOptions, CDefaultOptions } from '../../../common/interface.options';
+import { ENotificationType } from 'chipmunk.client.toolkit';
+
+interface Irgb {
+    red: number;
+    green: number;
+    blue: number;
+    opacity: number;
+}
+
+const LIMIT = 500;
+
+@Component({
+    selector: 'lib-dia-port-available-com',
+    templateUrl: './template.html',
+    styleUrls: ['./styles.less']
+})
+
+export class DialogAvailablePortComponent implements OnDestroy, AfterViewInit, OnInit {
+
+    @Input() port: IPortInfo;
+    @Input() observables: {tick: Subject<boolean>, resize: Subject<{ sidebar_width: number, sidebar_height: number }>};
+
+    @ViewChildren('canvas') canvases: QueryList<ElementRef>;
+
```

Comment:
Why we need ViewChildren, if template could have only one canvas element? Probably better just ViewChild?